failing cftime upstream-dev tests

Another set of failing `cftime` [tests](https://dev.azure.com/xarray/xarray/_build/results?buildId=2535&view=logs&jobId=2280efed-fda1-53bd-9213-1fa8ec9b4fa8&j=2280efed-fda1-53bd-9213-1fa8ec9b4fa8&t=175181ee-1928-5a6b-f537-168f7a8b7c2d), this time because of warnings while calling `open_dataset` (I think?):
```
FAILED xarray/tests/test_backends.py::test_use_cftime_standard_calendar_default_in_range[gregorian]
FAILED xarray/tests/test_backends.py::test_use_cftime_standard_calendar_default_in_range[proleptic_gregorian]
FAILED xarray/tests/test_backends.py::test_use_cftime_standard_calendar_default_in_range[standard]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[1500-360_day]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[1500-standard]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2000-360_day]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2000-365_day]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2000-366_day]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2000-all_leap]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2000-gregorian]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2000-julian]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2000-noleap]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2000-proleptic_gregorian]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2000-standard]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2500-360_day]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2500-365_day]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2500-366_day]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2500-all_leap]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2500-gregorian]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2500-julian]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2500-noleap]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2500-proleptic_gregorian]
FAILED xarray/tests/test_backends.py::test_use_cftime_true[2500-standard]
FAILED xarray/tests/test_backends.py::test_use_cftime_false_standard_calendar_in_range[gregorian]
FAILED xarray/tests/test_backends.py::test_use_cftime_false_standard_calendar_in_range[proleptic_gregorian]
FAILED xarray/tests/test_backends.py::test_use_cftime_false_standard_calendar_in_range[standard]
```

```pytb
________ test_use_cftime_standard_calendar_default_in_range[gregorian] _________

calendar = 'gregorian'
        for v in ["x", "time"]:
            original[v].attrs["units"] = units
            original[v].attrs["calendar"] = calendar
    
        x_timedeltas = np.array(x).astype("timedelta64[D]")
        time_timedeltas = np.array(time).astype("timedelta64[D]")
        decoded_x = np.datetime64(units_date, "ns") + x_timedeltas
        decoded_time = np.datetime64(units_date, "ns") + time_timedeltas
        expected_x = DataArray(decoded_x, [("time", decoded_time)], name="x")
        expected_time = DataArray(decoded_time, [("time", decoded_time)], name="time")
    
        with create_tmp_file() as tmp_file:
            original.to_netcdf(tmp_file)
            with pytest.warns(None) as record:
                with open_dataset(tmp_file, use_cftime=False) as ds:
                    assert_identical(expected_x, ds.x)
                    assert_identical(expected_time, ds.time)
>               assert not record
E               assert not WarningsChecker(record=True)

xarray/tests/test_backends.py:4378: AssertionError

```

--- a/xarray/tests/test_backends.py
+++ b/xarray/tests/test_backends.py
@@ -4334,6 +4334,12 @@ def test_source_encoding_always_present():
             assert ds.encoding["source"] == tmp
 
 
+def _assert_no_dates_out_of_range_warning(record):
+    undesired_message = "dates out of range"
+    for warning in record:
+        assert undesired_message not in str(warning.message)
+
+
 @requires_scipy_or_netCDF4
 @pytest.mark.parametrize("calendar", _STANDARD_CALENDARS)
 def test_use_cftime_standard_calendar_default_in_range(calendar):
@@ -4360,7 +4366,7 @@ def test_use_cftime_standard_calendar_default_in_range(calendar):
             with open_dataset(tmp_file) as ds:
                 assert_identical(expected_x, ds.x)
                 assert_identical(expected_time, ds.time)
-            assert not record
+            _assert_no_dates_out_of_range_warning(record)
 
 
 @requires_cftime
@@ -4423,7 +4429,7 @@ def test_use_cftime_true(calendar, units_year):
             with open_dataset(tmp_file, use_cftime=True) as ds:
                 assert_identical(expected_x, ds.x)
                 assert_identical(expected_time, ds.time)
-            assert not record
+            _assert_no_dates_out_of_range_warning(record)
 
 
 @requires_scipy_or_netCDF4
@@ -4452,7 +4458,7 @@ def test_use_cftime_false_standard_calendar_in_range(calendar):
             with open_dataset(tmp_file, use_cftime=False) as ds:
                 assert_identical(expected_x, ds.x)
                 assert_identical(expected_time, ds.time)
-            assert not record
+            _assert_no_dates_out_of_range_warning(record)
 
 
 @requires_scipy_or_netCDF4

